### PR TITLE
Removed some redundant else branches and cleaned up imports

### PR DESCRIPTION
**Pull request by montlikadani**
kép

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/EssentialsConf.java [hunk 1]
```diff
@@ -392,11 +392,11 @@ public void setProperty(final String path, final ItemStack stack) {
         set(path, map);
     }
 
-    public void setProperty(String path, List object) {
+    public void setProperty(String path, List<?> object) {
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/EssentialsConf.java]: Never seen this used before. Is it something in a newer version of Java that the IDE complains about now?

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 2]
```diff
@@ -494,8 +494,12 @@ public void onPlayerCommandPreprocess(final PlayerCommandPreprocessEvent event)
             switch (pluginCommand.getName()) {
                 case "afk":
                     update = false;
+                    break;
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java]: Are you certain that adding these breaks doesn't change behaviour here?

File: Essentials/src/com/earth2me/essentials/commands/Commandnuke.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ protected void run(final Server server, final CommandSource sender, final String
         if (args.length > 0) {
             targets = new ArrayList<>();
             int pos = 0;
-            for (String arg : args) {
+            for (int i = 0; i < args.length; i++) {
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandnuke.java]: Why?

**Review by pop4959 — CHANGES_REQUESTED** [review]

**Comment by pop4959** [comment]
I didn't leave too many comments because most of it was all the same kind of thing.

I'm not really sure I agree with removing the else branches because I think it does hurt readability quite a bit. It's fine in places where the function ends and it's an obvious catch-all, but in other places I think one can easily disassociate what used to be the else case with other code, making it less clear. I think I'd like to have @md678685's opinion on this as well though, especially since this PR comes mostly down to code style.

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java (hunk 2 above)
Comment (Ichbinjoe) [reply Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java]: This changes the behavior. Unless you can show that this is causing a bug, revert this change.

File: Essentials/src/com/earth2me/essentials/EssentialsConf.java [hunk 4]
```diff
@@ -127,9 +127,9 @@ public synchronized void load() {
                     result = decoder.decode(buffer, data, true);
                     if (result.isError()) {
                         throw new InvalidConfigurationException("Invalid Characters in file " + configFile.getAbsolutePath());
-                    } else {
-                        decoder.flush(data);
                     }
+
+                    decoder.flush(data);
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/EssentialsConf.java]: If you are going to do this, go all the way and unnest the else below!

File: Essentials/src/com/earth2me/essentials/EssentialsConf.java (hunk 1 above)
Comment (Ichbinjoe) [reply Essentials/src/com/earth2me/essentials/EssentialsConf.java]: It makes List 'typed' with nothingness. I guess its helpful?

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 5]
```diff
@@ -616,6 +620,7 @@ public void onPlayerInteract(final PlayerInteractEvent event) {
                         break;
                     }
                 }
+                break;
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java]: This changes the behavior. Revert.

File: Essentials/src/com/earth2me/essentials/PlayerList.java [hunk 6]
```diff
@@ -40,7 +40,7 @@ public static String listSummary(final IEssentials ess, final User user, final b
         for (User onlinePlayer : ess.getOnlineUsers()) {
             if (onlinePlayer.isHidden() || (user != null && !user.getBase().canSee(onlinePlayer.getBase()))) {
                 playerHidden++;
-                if (showHidden || user.getBase().canSee(onlinePlayer.getBase())) {
+                if (showHidden || user != null && user.getBase().canSee(onlinePlayer.getBase())) {
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/PlayerList.java:43]: Bug fix! Nice!

File: Essentials/src/com/earth2me/essentials/SpawnMob.java [hunk 7]
```diff
@@ -154,7 +154,9 @@ private static void spawnMob(final IEssentials ess, final Server server, final C
                     changeMobData(sender, mMob.getType(), spawnedMount, data.get(next).toLowerCase(Locale.ENGLISH), target);
                 }
 
-                spawnedMob.setPassenger(spawnedMount);
+                if (spawnedMob != null) {
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/SpawnMob.java]: `spawnedMob` is never null (or bad things happen). Revert change.

If you want to fix the compiler madness, rewrite this so its not as bad.

File: Essentials/src/com/earth2me/essentials/commands/Commanddelhome.java [hunk 8]
```diff
@@ -60,9 +59,9 @@ public void run(final Server server, final CommandSource sender, final String co
         if (args.length == 1) {
             if (canDelOthers) {
                 return getPlayers(server, sender);
-            } else {
-                return user.getHomes();
             }
+
+            return user != null ? user.getHomes() : Collections.emptyList();
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/commands/Commanddelhome.java]: `user` is never null - if `user` is null, `canDelOthers` is true and returns on 61.

File: Essentials/src/com/earth2me/essentials/commands/Commandinvsee.java [hunk 9]
```diff
@@ -37,11 +37,10 @@ protected void run(final Server server, final User user, final String commandLab
     protected List<String> getTabCompleteOptions(Server server, User user, String commandLabel, String[] args) {
         if (args.length == 1) {
             return getPlayers(server, user);
-        } else {
-            //if (args.length == 2) {
-            //    return Lists.newArrayList("equipped");
-            //}
-            return Collections.emptyList();
         }
+        //if (args.length == 2) {
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/commands/Commandinvsee.java]: Remove commented section - if its commented code, it should be deleted code.

File: Essentials/src/com/earth2me/essentials/commands/Commandnuke.java (hunk 3 above)
Comment (Ichbinjoe) [reply Essentials/src/com/earth2me/essentials/commands/Commandnuke.java]: If you are trying to remove the pos environment variable, you should uh, _actually remove it_.

File: Essentials/src/com/earth2me/essentials/messaging/SimpleMessageRecipient.java [hunk 10]
```diff
@@ -80,6 +80,7 @@ public String getName() {
                 } else {
                     sendMessage(tl("userAFK", recipient.getDisplayName()));
                 }
+            break;
```
Comment (Ichbinjoe) [inline Essentials/src/com/earth2me/essentials/messaging/SimpleMessageRecipient.java]: This introduces a bug. Revert.

**Review by Ichbinjoe — CHANGES_REQUESTED** [review]
Is there a tool which can perform all of these style fixes? I don't feel comfortable since this obviously was performed semi by hand (considering there were a few instances where inserted breaks would cause a bug). There was one bug fix, so that should be included!

**Comment by Ichbinjoe** [comment]
The diff viewer is showing entire files added / removed. I'm not going to review this until this PR is split up better - there is just _so many changes_.

**Comment by montlikadani** [comment]
I hope this correct now.

**Comment by mdcfe** [comment]
I'd rather we implemented an automated tool to check and optionally correct code style such as Checkstyle and mandate that PR code style matches an actual set of rules, rather than just relying on individual personal judgements.

I have a branch locally on which I ran Checkstyle + automatic corrections, so that would be much faster than manually going through every class.

**Comment by montlikadani** [comment]
Closed due to many attempts and other errors. @md678685  will find an easier way to do this.

For NPE checking: Will open a new PR

**Pull request closed without merging**